.test-infra/mock-apis/go.mod

```diff
@@ -21,7 +21,7 @@
 module github.com/apache/beam/test-infra/mock-apis
 
 go 1.21
-toolchain go1.24.1
```

Comment:
Would this bump the required go version for users. If it does we should refrain from doing it too frequently (context: https://github.com/apache/beam/pull/31567#discussion_r1635382434)

Reply:
Anyway, it fixes Go download issue which is critical for now